### PR TITLE
Fix pod logs issue when pod had no proxy

### PR DESCRIPTION
**Pull request by jshaughn**
Also
- Fix issue when pod had no app container
- Fix issue when app container had no version label
  - and use configured version label name

https://github.com/kiali/kiali/issues/3388

### CODE REVIEW

**Review by hhovsepy — APPROVED** [review]
Verified:
Screenshot from 2020-11-03 09-54-43

**Comment by lucasponce** [comment]
```
> Failed to compile.
./src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx
  Line 748:26:  Expected to return a value in arrow function  array-callback-return
```

It needs a minor update to please travis.

File: src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx [hunk 1]
```diff
@@ -735,21 +734,22 @@ export default class WorkloadPodLogs extends React.Component<WorkloadPodLogsProp
...
+    containers = containers.filter(c => c.name !== 'istio-proxy');
+    const options: { [key: string]: string } = {};
+
+    if (containers.length === 0) {
+      options[NoAppContainer] = NoAppContainer;
+      return { container: NoAppContainer, containerOptions: options };
+    }
+
+    const containerNames = containers.map(c => c.name);
+    containerNames.map(n => {
```
Comment (cfcosta) [inline src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx]: Nitpick: Shouldn't this be a reduce? At least a `forEach`.

File: src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx [hunk 2]
```diff
@@ -862,9 +859,9 @@ export default class WorkloadPodLogs extends React.Component<WorkloadPodLogsProp
     });
   };
 
-  private entryToString = (entries: LogEntry[]): string => {
+  private entriesToString = (entries: LogEntry[]): string => {
     return entries.map(le => (this.state.showTimestamps ? `${le.timestamp} ${le.message}` : le.message)).join('\n');
   };
 
-  private hasEntries = (entries: LogEntry[]): boolean => entries.length > 0;
+  private hasEntries = (entries: LogEntry[]): boolean => !!entries && entries.length > 0;
```
Comment (cfcosta) [inline src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx:866]: No need for the `||`, unless you set `entries` to be `LogEntry[] || undefined`.

**Review by cfcosta — APPROVED** [review]
You added some changes while I was reviewing :)
Should be ok if travis agree with me.

File: src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx (hunk 1 above)
Comment (jshaughn) [reply src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx]: yes, I changed it to forEach.

File: src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx (hunk 2 above)
Comment (jshaughn) [reply src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx:866]: I originally did not have the truthy test and for whateve reason this code was being reached with entries as undefined.  It's possible other refactors fixed that issue, but leaving as it hurts nothing, except your eyes :)

**Pull request merged**